src/pages/index.js

```diff
@@ -0,0 +1,328 @@
+    avatarElement.src = userInfo.avatar;
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+//modal elements
+const modals = document.querySelectorAll(".modal");
+const avatarElement = document.querySelector(".profile__avatar");
+const profileNameElement = document.querySelector(".profile__name");
+const profileJobElement = document.querySelector(".profile__description");
+const profileEditButton = document.querySelector(".profile__button-secondary");
+const profilePostButton = document.querySelector(".profile__button-primary");
+
+//Edit Elements
+const editModal = document.querySelector("#edit-modal");
+const editModalCloseButton = editModal.querySelector(".modal__button");
+const editModalSubmitButton = editModal.querySelector(
+  "#modal__edit-submit-button"
+);
+const editFormElement = editModal.querySelector("#modal__edit-form");
+const editNameInput = editModal.querySelector("#profile-name");
+const editJobInput = editModal.querySelector("#profile-description");
+
+//Post Elements
+const postModal = document.querySelector("#post-modal");
+const postModalCloseButton = postModal.querySelector(".modal__button");
+const postFormElement = postModal.querySelector("#modal__post-form");
+const postModalSubmitButton = postModal.querySelector(
+  "#modal__post-submit-button"
+);
+const postLinkInput = postModal.querySelector("#image-link");
+const postCaptionInput = postModal.querySelector("#newpost-caption");
+
+//Preview Card Image Elements
+const previewModal = document.querySelector("#preview-modal");
+const previewCloseButton = previewModal.querySelector(
+  ".modal__close-icon_type_preview"
+);
+const previewModalImage = previewModal.querySelector(".modal__image");
+const previewModalCaption = previewModal.querySelector(".modal__caption");
+
+//Delete confimation Elements
+const deleteModal = document.querySelector("#delete-modal");
+const deleteCloseButton = deleteModal.querySelector("#close-icon-delete");
+const deleteCloseImage = deleteModal.querySelector("#delete-close-icon");
+const deleteCancelButton = deleteModal.querySelector(
+  ".modal__delete-type-cancel"
+);
+const deleteForm = deleteModal.querySelector(".modal__button-container");
+
+//card element
+const cardTemplate = document.querySelector("#card-template");
+const cardList = document.querySelector(".cards");
+
+//Edit Avatar
+const avatarModal = document.querySelector("#avatar-modal");
+const avatarModalCloseButton = avatarModal.querySelector(".modal__close-icon");
+const avatarForm = avatarModal.querySelector(".modal__form");
+const avatarSubmitButton = avatarModal.querySelector(".modal__button-avatar");
+const avatarInput = avatarModal.querySelector("#profile-avatar-input");
+const avatarModalButton = document.querySelector(".profile__avatar-button");
+
+let selectedCard;
+let selectedCardId;
+
+function handleProfileFormSubmit(evt) {
+  evt.preventDefault();
+  const submitbtn = evt.submitter;
+  submitbtn.textContent = "Saving...";
```

Comment:
**COULD BE IMPROVED**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **resetting** and **catching errors**


```js
// define a function for changing the button text. It accepts 4 params (the 2 last are optional with default texts)
export function renderLoading(isLoading, button, buttonText='Save', loadingText='Saving...') {
  if (isLoading) {
    button.textContent = loadingText
  } else {
    button.textContent = buttonText
  }
}

// define a universal function that accepts a request function, event and a default loading text 
function handleSubmit(request, evt, loadingText = 'Saving...') {
 // You need to prevent the default action in any submit handler
  evt.preventDefault();

  // the button is always available inside `event` as `submitter`
  const submitButton = evt.submitter;
  // fix the initial button text
  const initialText = submitButton.textContent;
  // change the button text before requesting
  renderLoading(true, submitButton, initialText, loadingText);
  // call the request function to be able to use the promise chain
  request()
    .then(() => {
      // any form should be reset after a successful response 
      // evt.target is the form in any submit handler		
      evt.target.reset();
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
    .catch(console.error)      

    // and in finally we need to stop loading
    .finally(() => {
      renderLoading(false, submitButton, initialText);
    });
}

```

Here is an example of handling the profile form submit:

```js
function handleProfileFormSubmit(evt) {
  // create a request function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally`
    return editProfile(nameInput.value, jobInput.value).then((userData) => {
      userName.textContent = userData.name;
      userJob.textContent = userData.about;
    });
  }
  // here we call handleSubmit passing the request and event (if you want a different loading text then you need to pass the 3rd param)
  handleSubmit(makeRequest, evt);
}

```

So, this way you can remove a lot of code duplicating. You will not need to search buttons, pass initial button texts and so on.

`handleSubmit` and `renderLoading` should be placed in `utils.js`, because they are utility functions